### ISSUE TITLE
Support more formats (PNG…) for thumbnail and preview image

### ISSUE THREAD

**Issue by mmuman** [Chocobozzz/PeerTube#1953]
On low resolution videos (like, [capture of old computer emulators](https://video.g3l.org/videos/watch/7004799e-7564-4cba-90dd-cb190c1d8e01)), using jpeg for thumbnails makes them very blurry. Supporting PNG would fix this.

Even though it might be a wanted effect, it's not always the case. Example:
![vlcsnap-2019-07-02-04h05m46s703](https://user-images.githubusercontent.com/470143/60513828-6f8ea880-9cd8-11e9-86c7-ee3c985d28c1.jpg) vs ![treize](https://user-images.githubusercontent.com/470143/60513839-77e6e380-9cd8-11e9-9533-ef63f6f0bac7.png)

**Comment by Chocobozzz** [comment]
You mean when you upload a specific image for a video thumbnail? Or when PeerTube automatically created the thumbnail from the video?

We can't support png images for thumbnails because they are federated and it would create file size issues. But maybe we can improve the jpeg compression or ffmpeg thumbnail generation :thinking:

**Comment by mmuman** [comment]
I mean uploading an existing png file. I didn't try letting it make a thumbnail from the video, but I suspect it wouldn't be any better than a jpeg.

I don't really see the problem with PNG, the size really depends on the content, a simple graphic can probably be smaller in a PNG than JPEG.

**Comment by Chocobozzz** [comment]
> I don't really see the problem with PNG, the size really depends on the content, a simple graphic can probably be smaller in a PNG than JPEG.

Good point. We just have to check the PNG thumbnail file is not bigger than a limit (~10ko for example) after [sharp](https://sharp.pixelplumbing.com/en/stable/) resizing.

**Comment by notthewave** [comment]
https://github.com/Chocobozzz/PeerTube/issues/2280
similar issue

**Comment by aschaap** [comment]
What about WebP? Lossless WebP tends to be [significantly](https://cloudinary.com/blog/flif_the_new_lossless_image_format_that_outperforms_png_webp_and_bpg) smaller than PNG, and has become [well-supported](https://caniuse.com/webp) across browsers.

**Comment by notthewave** [comment]
Didn’t know webP could be lossless. I spin up ImageMagick real quick and add it to the comparison above.

**Comment by notthewave** [comment]
magick wizard.png -quality 50 -define webp:lossless=true wizard.webp
png (original)
webp whielded some amazing results:
![test](https://user-images.githubusercontent.com/74874782/113903964-10b96580-97c1-11eb-88fb-5b2e67586805.png)
webp lossless
https://0x0.st/-cU5.webp 
lol github does not support webp

png 5.4K
vs webp 1.8K

**Comment by notthewave** [comment]
ironically the jpeg above seems to be
**_27K_** big Lel

**Comment by Agorise** [comment]
WebP is awesome, especially little animated webp images. They load fast, look great and enable the designer to really do some eye-catching work.

**Comment by Chocobozzz** [comment]
Implemented in https://github.com/Chocobozzz/PeerTube/commit/526f1c6df529fa7620970c216f8915707b9f755e

PeerTube will keep the input file image format now
